File: Python solutions/1290. Convert Binary Number in a Linked List to Integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def binary_to_decimal(self, s: str) -> int:
        decimal = 0
        for digit in s:
            decimal = (decimal * 2) + int(digit)
        return decimal
        
    def getDecimalValue(self, head: ListNode) -> int:
        s = ""
        while head:
            s += str(head.val)
            head = head.next
        return self.binary_to_decimal(s)

File: C++ solutions/1290. Convert Binary Number in a Linked List to Integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int binary_to_decimal(string s){
        int decimal = 0;
        for(int i=0; i<s.size();i++){
            decimal = (decimal * 2) + (s[i]-'0');
        }
        return decimal;
    }

    int getDecimalValue(ListNode* head) {
        string s = "";
        while (head != nullptr) {
            s += to_string(head->val);
            head = head->next;
        }
        return binary_to_decimal(s);
    }
};